types/src/binary_port/type_wrappers.rs

```diff
@@ -298,6 +300,50 @@ impl FromBytes for TransactionWithExecutionInfo {
     }
 }
 
+/// A query result for a dictionary item, contains the dictionary item key and a global state query result.
+#[derive(Debug)]
+pub struct DictionaryQueryResult {
+    key: Key,
+    query_result: GlobalStateQueryResult,
+}
+
+impl DictionaryQueryResult {
+    /// Constructs new dictionary query result.
+    pub fn new(key: Key, query_result: GlobalStateQueryResult) -> Self {
+        Self { key, query_result }
+    }
+
+    /// Converts `self` into the dictionary item key and global state query result.
+    pub fn into_inner(self) -> (Key, GlobalStateQueryResult) {
+        (self.key, self.query_result)
+    }
+}
+
+impl ToBytes for DictionaryQueryResult {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        let mut buffer = bytesrepr::allocate_buffer(self)?;
+        self.write_bytes(&mut buffer)?;
+        Ok(buffer)
+    }
+
+    fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), bytesrepr::Error> {
+        self.key.write_bytes(writer)?;
+        self.query_result.write_bytes(writer)
+    }
+
+    fn serialized_length(&self) -> usize {
+        self.key.serialized_length() + self.query_result.serialized_length()
+    }
+}
+
+impl FromBytes for DictionaryQueryResult {
```

Comment:
Please consider adding `dictionary_query_result_roundtrip` test. Could be worth checking if other types that were recently added or modified are covered with roundtrip tests.

Reply:
good catch, one other test was missing as well, I've added it